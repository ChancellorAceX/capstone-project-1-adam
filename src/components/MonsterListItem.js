import React from 'react';
import '../styles/MonsterListItem.css';
import { Link } from 'react-router-dom';
// import ApiContext from './ApiContext';

export default function MonsterListItem(props) {

  return (
    <Link to={`/bestiary/${props.monster.bid}`} style={{ textDecoration: 'none' }}>
      <li className='monsterListing'>
        {/* possibly add an 'add to' button here */}
        <p className='name'>{props.monster.monstername}</p>
        <section className='type'>
          <p className='type'>{props.monster.type.split(', ')[0]}</p>
          <p className='type'>{props.monster.type.split(', ')[1]}</p>
        </section>
        <p className='cr'>CR: {props.monster.cr}</p>
      </li>
    </Link>
  );
}
