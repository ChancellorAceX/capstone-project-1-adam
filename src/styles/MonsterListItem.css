* {
  box-sizing: border-box;
}

li.monsterListing {
  height:fit-content;
  min-height:25px;
  width:100vw;
  display:flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border:1px solid gray;
  padding:5px;
  background-color: black;
}

li.monsterListing p {
  margin:0;
  color:white;
}

p.name{
  width:30vw;
  padding-left: 5px;
  flex-shrink:3;
}

p.type{
  width:40vw;
  flex-shrink:1;
}

p.cr {
  width:20vw;
  flex-shrink:5
}