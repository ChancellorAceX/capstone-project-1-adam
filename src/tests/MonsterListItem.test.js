import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MonsterListItem from '../components/MonsterListItem';

describe('MonsterListItem component', () => {
const props = {
  "monster": {
    "bid": 5,
    "monstername": "Solar",
    "type": "Large celestial, lawful good",
    "ac": "21 (natural armor)",
    "maxhp": "18d10 + 144",
    "speed": 50,
    "burrowspeed": null,
    "climbspeed": null,
    "flyspeed": 150,
    "swimspeed": null,
    "str": 26,
    "strmod": 8,
    "dex": 22,
    "dexmod": 6,
    "con": 26,
    "conmod": 8,
    "int": 25,
    "intmod": 7,
    "wis": 25,
    "wismod": 7,
    "cha": 30,
    "chamod": 10,
    "savingthrows": "Int +14, Wis +14, Cha +17",
    "skills": "Perception +14",
    "vulnerabilities": null,
    "resistances": "radiant; bludgeoning, piercing, and slashing from nonmagical weapons",
    "immunities": "necrotic, poison, charmed, exhaustion, frightened, poisoned",
    "senses": "Truesight 120 ft., passive Perception 24",
    "languages": "all, telepathy 120 ft.",
    "cr": 21,
    "xp": 33000,
    "extras": "[Array(2), Array(2), Array(2), Array(2)]",
    "actions": "[Array(2), Array(2), Array(2), Array(2), Array(2), â€¦]"
  }
};

  it('renders the full listing component', () => {
    const wrapper = shallow(<MonsterListItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});