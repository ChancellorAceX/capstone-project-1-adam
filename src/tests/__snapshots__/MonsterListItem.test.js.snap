// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonsterListItem component renders the full listing component 1`] = `
<Link
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/bestiary/5"
>
  <li
    className="monsterListing"
  >
    <p
      className="name"
    >
      Solar
    </p>
    <section
      className="type"
    >
      <p
        className="type"
      >
        Large celestial
      </p>
      <p
        className="type"
      >
        lawful good
      </p>
    </section>
    <p
      className="cr"
    >
      CR: 
      21
    </p>
  </li>
</Link>
`;
